### PR TITLE
More fixups of SvPileup; transition to using Metrics class; cleanup o…

### PR DESCRIPTION
**Pull request by tfenne**
…f FilterAndMerge

### CODE REVIEW

File: src/main/scala/com/fulcrumgenomics/sv/tools/FilterAndMerge.scala [hunk 1]
```diff
@@ -33,166 +31,106 @@ class FilterAndMerge
...
-      .sortBy { p =>
-        (dict(p.left_contig).index, dict(p.right_contig).index, p.left_start, p.left_end, p.right_start, p.right_end)
-      }
-      .foreach { merged =>
-        val values = MergedPileup.values(merged)
-        writer.write(values.mkString("\t"))
-        writer.write('\n')
-      }
-    writer.close()
+    Metric.write(output, results.result().sorted)
```
Comment (nh13) [inline src/main/scala/com/fulcrumgenomics/sv/tools/FilterAndMerge.scala:52]: changes the sort :/

File: src/main/scala/com/fulcrumgenomics/sv/tools/SvPileup.scala [hunk 2]
```diff
@@ -91,35 +99,33 @@ class SvPileup
...
-      .sortWith(Breakpoint.PairedOrdering.lt)
+  /** Write the breakpoints to a file. */
+  private def writeBreakpoints(path: FilePath,
+                               tracker: BreakpointTracker,
+                               dict: SequenceDictionary): Unit = {
+    val writer      = Metric.writer[BreakpointPileup](path)
+    val breakpoints = tracker.breakpoints.toIndexedSeq.sortWith(Breakpoint.PairedOrdering.lt)
 
     breakpoints.foreach { bp =>
       val info         = tracker(bp)
```
Comment (nh13) [inline src/main/scala/com/fulcrumgenomics/sv/tools/SvPileup.scala:110]:
```suggestion
      val info       = tracker(bp)
```

File: src/main/scala/com/fulcrumgenomics/sv/tools/SvPileup.scala [hunk 3]
```diff
@@ -91,35 +99,33 @@ class SvPileup
...
-      val id           = info.id
-      val leftRefName  = dict(bp.leftRefIndex).name
-      val rightRefName = dict(bp.rightRefIndex).name
-
-      val values = Iterator(
-        id, leftRefName, bp.leftPos, toStrand(bp.leftPositive), rightRefName, bp.rightPos, toStrand(bp.rightPositive),
-        info.splitRead, info.readPair, info.total
+      val metric       = BreakpointPileup(
```
Comment (nh13) [inline src/main/scala/com/fulcrumgenomics/sv/tools/SvPileup.scala:111]:
```suggestion
      val metric = BreakpointPileup(
```

**Review by nh13 — APPROVED** [review]

File: src/main/scala/com/fulcrumgenomics/sv/tools/FilterAndMerge.scala (hunk 1 above)
Comment (tfenne) [reply src/main/scala/com/fulcrumgenomics/sv/tools/FilterAndMerge.scala:52]: Well the old sort involved the `sameStrand` which I don't think makes too much sense any more.  If you have a different preference for how to order fields I'm game - I thought I did this in a way that would give you the closest to what you had before (it's using the ordering on the new metric class, not Breakpoint).

**Pull request merged**